mysql/mysqldump.go

```diff
@@ -342,6 +342,14 @@ func updateColsByOption(conv *internal.Conv, tableName string, col *ast.ColumnDe
 			column.NotNull = true
 		case ast.ColumnOptionDefaultValue:
 			column.Ignored.Default = true
+			// If a data type specification includes no explicit DEFAULT
+			// value, MySQL determines if the column can take NULL as a value
+			// and the column is defined with DEFAULT NULL clause in mysqldump.
+			// This case is ignored from issue reporting of 'Deafult' value.
+			defaultValue := fmt.Sprintf("%v", elem.Expr.(*driver.ValueExpr).GetValue())
```

Comment:
Probably best to avoid the type assertion here (which could trigger panic), and instead use the "ok" version so we can test the assertion.
Also let's avoid the Sprintf and do direct value comparison.
i.e. elem.Expr.(*driver.ValueExpr).GetValue() == nil.
And simplest to make the assignment to true conditional, instead of having two assignments.

How about:

			v, ok := elem.Expr.(*driver.ValueExpr)
			if !ok || v.GetValue() != nil {
				column.Ignored.Default = true
			}

or maybe a bit more readable as:

			// If a data type specification includes no explicit DEFAULT
			// value, MySQL determines if the column can take NULL as a value
			// and the column is defined with DEFAULT NULL clause in mysqldump.
			// This case is ignored from issue reporting of 'Default' value.
  			v, ok := elem.Expr.(*driver.ValueExpr)
                        nullDefault := ok && v.GetValue() == nil
                        if !nullDefault {
                                column.Ignored.Default = true
                        }

Reply:
I like the second one.

Reply:
Sure, updated according to second one.